src/Xrefcheck/Verify.hs

```diff
@@ -417,11 +420,30 @@ verifyReference
       unless (isVirtual referredFile) do
         checkReferredFileIsInsideRepo referredFile
         checkReferredFileExists referredFile
-        case M.lookup referredFile repoInfo of
+        case lookupFilePath referredFile $ M.toList repoInfo of
           Nothing -> pass  -- no support for such file, can do nothing
           Just referredFileInfo -> whenJust mAnchor $
             checkAnchor referredFile (_fiAnchors referredFileInfo)
 
+    lookupFilePath :: FilePath -> [(FilePath, Maybe FileInfo)] -> Maybe FileInfo
+    lookupFilePath fp = snd <=< find (equalFilePath (expandIndirections fp) . fst)
+
+    -- expands ".." and "."
+    -- expandIndirections "a/b/../c"      = "a/c"
+    -- expandIndirections "a/b/c/../../d" = "a/d"
+    -- expandIndirections "../../a"       = "../../a"
+    -- expandIndirections "a/./b"         = "a/b"
+    -- expandIndirections "a/b/./../c"    = "a/c"
+    expandIndirections :: FilePath -> FilePath
```

Comment:
As far as I can tell, this is needed because otherwise `dir1/dir2/..` won't match with `dir1`, even tho it should, right?

It works, but it does feel a bit hackish :thinking: 

I think we should solve this problem by using canonical paths across the entire codebase though. Then, simply using `==` everywhere should work.

This will probably be a bigger effort, so I'll write a separate issue for this.

Reply:
Okay, created a follow-up issue here: https://github.com/serokell/xrefcheck/issues/197

Reply:
Yes, you got it right. Thanks for creating the issue.